llvm/lib/Transforms/Scalar/LoopStrengthReduce.cpp

```diff
@@ -6830,15 +6830,18 @@ canFoldTermCondOfLoop(Loop *L, ScalarEvolution &SE, DominatorTree &DT,
     return std::nullopt;
   }
 
-  if (!SE.hasLoopInvariantBackedgeTakenCount(L)) {
-    LLVM_DEBUG(dbgs() << "Cannot fold on backedge that is loop variant\n");
-    return std::nullopt;
-  }
-
   BasicBlock *LoopLatch = L->getLoopLatch();
   BranchInst *BI = dyn_cast<BranchInst>(LoopLatch->getTerminator());
   if (!BI || BI->isUnconditional())
     return std::nullopt;
+
+  if (!SE.hasLoopInvariantBackedgeTakenCount(L)) {
+    if (isa<SCEVCouldNotCompute>(SE.getExitCount(L, BI->getParent()))) {
```

Comment:
This change has nothing to do with the patch description.  Extending term folding to multiple exit loops is quite subtle, and will absolutely require it's own review.

Reply:
Sorry, I update the description about it and invite it's owner to review. Based on my limited understanding, if the exit count of LoopLatch is computable and the condition just has one use, it's safe to fold it.